cli/azd/internal/appdetect/java.go

```diff
@@ -43,165 +39,55 @@ func (jd *javaDetector) DetectProject(ctx context.Context, path string, entries
-	Properties           Properties           `xml:"properties"`
-	Dependencies         []dependency         `xml:"dependencies>dependency"`
-	DependencyManagement dependencyManagement `xml:"dependencyManagement"`
-	Build                build                `xml:"build"`
-	path                 string
-}
-
-// Parent represents the parent POM if this project is a module.
-type parent struct {
-	GroupId    string `xml:"groupId"`
-	ArtifactId string `xml:"artifactId"`
-	Version    string `xml:"version"`
-}
-
-type Properties struct {
-	Entries []Property `xml:",any"` // Capture all elements inside <properties>
-}
-
-type Property struct {
-	XMLName xml.Name
-	Value   string `xml:",chardata"`
-}
-
-// Dependency represents a single Maven dependency.
-type dependency struct {
-	GroupId    string `xml:"groupId"`
-	ArtifactId string `xml:"artifactId"`
-	Version    string `xml:"version"`
-	Scope      string `xml:"scope,omitempty"`
-}
-
-// DependencyManagement includes a list of dependencies that are managed.
-type dependencyManagement struct {
-	Dependencies []dependency `xml:"dependencies>dependency"`
-}
-
-// Build represents the build configuration which can contain plugins.
-type build struct {
-	Plugins []plugin `xml:"plugins>plugin"`
-}
-
-// Plugin represents a build plugin.
-type plugin struct {
-	GroupId    string `xml:"groupId"`
-	ArtifactId string `xml:"artifactId"`
-	Version    string `xml:"version"`
-}
-
-func readMavenProject(filePath string) (*mavenProject, error) {
-	bytes, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, err
-	}
-
-	var initialProject mavenProject
-	if err := xml.Unmarshal(bytes, &initialProject); err != nil {
-		return nil, fmt.Errorf("parsing xml: %w", err)
-	}
-
-	// replace all placeholders with properties
-	str := replaceAllPlaceholders(initialProject, string(bytes))
-
-	var project mavenProject
-	if err := xml.Unmarshal([]byte(str), &project); err != nil {
-		return nil, fmt.Errorf("parsing xml: %w", err)
-	}
-
-	project.path = filepath.Dir(filePath)
-
-	return &project, nil
-}
-
-func replaceAllPlaceholders(project mavenProject, input string) string {
-	propsMap := parseProperties(project.Properties)
-
-	re := regexp.MustCompile(`\$\{([A-Za-z0-9-_.]+)}`)
-	return re.ReplaceAllStringFunc(input, func(match string) string {
-		// Extract the key inside ${}
-		key := re.FindStringSubmatch(match)[1]
-		if value, exists := propsMap[key]; exists {
-			return value
-		}
-		return match
-	})
-}
-
-func parseProperties(properties Properties) map[string]string {
-	result := make(map[string]string)
-	for _, entry := range properties.Entries {
-		result[entry.XMLName.Local] = entry.Value
-	}
-	return result
-}
-
-func detectDependencies(currentRoot *mavenProject, mavenProject *mavenProject, project *Project) (*Project, error) {
-	detectAzureDependenciesByAnalyzingSpringBootProject(currentRoot, mavenProject, project)
-	return project, nil
```

Comment:
The error will always been `nil`, so `return error` is not necessary. Deleted it in current PR.